truffle/src/main/ruby/core/array.rb

```diff
@@ -359,12 +362,12 @@ def fill_internal(a=undefined, b=undefined, c=undefined)
       right += 1 unless one.exclude_end?
       return self if right <= left           # Nothing to modify
 
-    elsif one and !undefined.equal?(one)
+    elsif one
       left = Rubinius::Type.coerce_to_collection_length one
       left += size if left < 0
       left = 0 if left < 0
 
-      if two and !undefined.equal?(two)
+      if !undefined.equal?(two) and two
```

Comment:
Convention is to add parenthesis `if (a = true) && b.nil?` to make the assignment stand out more. I like `test or raise something` form since it puts the test first not the unimportant exception as in `raise something unless test`. Anyway this is not important, it's opinionated and gain is just minor readability improvements.